spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -2582,16 +2585,35 @@ void ParserBuilder::skip(Expression* size, const Meta& location) {
 }
 
 void ParserBuilder::advanceToNextData() {
-    auto offset = builder()->memberCall(state().cur, "offset");
-    auto profiler = builder()->startProfiler(hilti::util::fmt("spicy/unit/%s/__gap__", state().unit_id), offset);
+    auto old_offset = builder()->addTmp("old_offset", builder()->memberCall(state().cur, "offset"));
+
+    auto profiler =
+        builder()->startProfiler(hilti::util::fmt("spicy/unit/%s/__sync_advance__", state().unit_id), old_offset);
 
     builder()->addAssign(state().cur, builder()->memberCall(state().cur, "advance_to_next_data"));
```

Comment:
The caveat here is that we will not call the hook exactly after `%sync-advance-block-size` bytes if that would mean calling from inside a gap chunk (instead the hook is called only on data chunks). This might be surprising and pretty subtle, but is likely inconsequential since dealing with gap chunks is cheap.

If we wanted to deal with this on a more granular level, we could call the hook in data chunks, or on chunk boundaries (so that a parser can react to only seeing gap chunks). This would require hooking into `advance_to_next_data`.

Reply:
Thinking about this more, it might be beneficial to document that we call the hook _earliest_ after `%sync-advance-block-size` bytes (i.e., current behavior, maybe just a docs tweak). With that we could then optimize advancing more freely in the future.

Reply:
Ok, I'll tweak the docs. I did mean to leave this somewhat fuzzy for easier/cheaper implementation, and because it shouldn't matter much.